benchmark/misc/restparams-bench.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+const common = require('../common.js');
+const assert = require('assert');
+const v8 = require('v8');
+
+const bench = common.createBenchmark(main, {
+  method: ['old', 'new', 'arguments'],
+  millions: [100]
+});
+
+function a() {
+  var len = arguments.length;
+  var args = new Array(len);
+  for (var i = 0; i < len; i++)
+    args[i] = arguments[i];
+  assert.strictEqual(args[0], 1);
+  assert.strictEqual(args[1], 2);
+  assert.strictEqual(args[2], 'a');
+  assert.strictEqual(args[3], 'b');
+}
+
+function b(...args) {
+  assert.strictEqual(args[0], 1);
+  assert.strictEqual(args[1], 2);
+  assert.strictEqual(args[2], 'a');
+  assert.strictEqual(args[3], 'b');
+}
+
+function c() {
+  assert.strictEqual(arguments[0], 1);
+  assert.strictEqual(arguments[1], 2);
+  assert.strictEqual(arguments[2], 'a');
+  assert.strictEqual(arguments[3], 'b');
+}
+
+function runA(n) {
+  a(1, 2, 'a', 'b');
+  v8.setFlagsFromString('--allow_natives_syntax');
+  eval('%OptimizeFunctionOnNextCall(a)');
+  a(1, 2, 'a', 'b');
+
+  var i = 0;
+  bench.start();
+  for (; i < n; i++)
+    a(1, 2, 'a', 'b');
+  bench.end(n / 1e6);
+}
+
+function runB(n) {
+  b(1, 2, 'a', 'b');
+  v8.setFlagsFromString('--allow_natives_syntax');
+  eval('%OptimizeFunctionOnNextCall(b)');
+  b(1, 2, 'a', 'b');
+
+  var i = 0;
+  bench.start();
+  for (; i < n; i++)
+    b(1, 2, 'a', 'b');
+  bench.end(n / 1e6);
+}
+
+function runC(n) {
+  c(1, 2, 'a', 'b');
+  v8.setFlagsFromString('--allow_natives_syntax');
+  eval('%OptimizeFunctionOnNextCall(c)');
+  c(1, 2, 'a', 'b');
+
+  var i = 0;
+  bench.start();
+  for (; i < n; i++)
+    b(1, 2, 'a', 'b');
```

Comment:
Also, I think `a()`, `b()`, and `c()` should be renamed to something more descriptive.